services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -493,13 +493,54 @@ specDeleteUser = do
                 !!! const 405 === statusCode
 
     describe "DELETE /Users/:id" $ do
+        it "should be delete successfully and be idempotent" $ do
+            (tok, _) <- registerIdPAndScimToken
+            user <- randomScimUser
+            storedUser <- createUser tok user
+            let uid = scimUserId storedUser
+
+            spar <- view teSpar
+            deleteUser_ (Just tok) (Just uid) spar
+                !!! const 204 === statusCode
+            deleteUser_ (Just tok) (Just uid) spar
+                !!! const 204 === statusCode
+
+
+        it "should return 401 if we don't provide a token" $ do
+            user <- randomScimUser
+            (tok, _) <- registerIdPAndScimToken
+            storedUser <- createUser tok user
+            spar <- view teSpar
+            let uid = scimUserId storedUser
+            deleteUser_ Nothing (Just uid) spar
+                !!! const 401 === statusCode
+
+        it "should return 403 if we provide a token for a different team" $ do
```

Comment:
"should return 404 if the user is in a different team"

Reply:
We chatted about this in the room; the decision we came to at the time was that at wire in general we haven't really tried to hide user ids; and if we aren't hiding them, then 403 gives a better user experience because it's much easier to understand and debug. Do you have a strong argument for 404 instead?

Reply:
My reasoning for 404 is that our use of unique IDs is an implementation detail and shouldn't affect the API status codes.

We could just as well have `/Users/1`, `/Users/2`, `/Users/3` which would be mapped to Brig IDs internally, and in that case you would definitely return 404, not 403.

This said, you can leave it as 403 if you change other endpoints to be consistent with that.